test/stablecoin_converter.js

```diff
@@ -902,6 +902,52 @@ contract("StablecoinConverter", async (accounts) => {
 
     await stablecoinConverter.submitSolution(batchIndex, owner, orderId, volume, prices, tokenIdsForPrice)
   })
+  it("checks that currentPrices between different solutions are reset", async () => {
+    const feeToken = await MockContract.new()
+    const stablecoinConverter = await StablecoinConverter.new(2 ** 16 - 1, feeDenominator, feeToken.address)
+    const erc20_2 = await MockContract.new()
+    const erc20_3 = await MockContract.new()
+
+    await feeToken.givenAnyReturnBool(true)
+    await erc20_2.givenAnyReturnBool(true)
+    await erc20_3.givenAnyReturnBool(true)
+
+
+    await stablecoinConverter.deposit(feeToken.address, 60000, { from: user_1 })
+    await stablecoinConverter.deposit(erc20_2.address, 60000, { from: user_2 })
+    await stablecoinConverter.deposit(erc20_3.address, 10000, { from: user_3 })
+
+    await stablecoinConverter.addToken(erc20_2.address)
+    await stablecoinConverter.addToken(erc20_3.address)
+
+    const batchIndex = (await stablecoinConverter.getCurrentStateIndex.call()).toNumber()
+
+    const orderId1 = await sendTxAndGetReturnValue(stablecoinConverter.placeOrder, 1, 0, true, batchIndex + 1, 59940, 60000, { from: user_1 })
```

Comment:
Not for now, but in the long run, I think we should refactor our testdata to come from a common module where each test cases picks there scenario (e.g. 3 orders, 2 solutions) or whatever it needs.